iwyu.cc

```diff
@@ -1875,6 +1875,12 @@ class IwyuBaseAstVisitor : public BaseAstVisitor<Derived> {
     if (IsFriendDecl(decl))
       return true;
 
+    // We ignore all uses of types in method declarations overriding a virtual
+    // function; these types are already provided by the base class header,
+    // which must be included by the derived.
+    if (IsOverriddenMethodDecl(decl))
```

Comment:
I am wondering why we need this check. All types in a method override decl will be ignored anyway, right?

Reply:
Good catch! It was necessary before for some half-broken removal of method bodies. Now that I do the is-definition check in the right place, it's no longer required. Thanks!

Reply:
I noticed while working on the improved handling of unused args that this was to suppress reporting of the return type in VisitFunctionDecl. It's also reported in VisitType. The new code semi-accidentally marks both cases ignored, but I'm starting to think it might be nicer to be explicit.